server/api/categories.mjs

```diff
@@ -0,0 +1,40 @@
+import { mongooseConnect } from "@/lib/mongoose";
+import { Category } from "@/models/Category";
+
+export default async function handle(req,res) {
+    const {method}= req;
+    await mongooseConnect();
+
+if (method === 'GET') {
+    res.json(await Category.find().populate('parent'))
+}
+
+
+    if (method === 'POST') {
+        const {name, parentCategory, properties} = req.body; 
+        const parent = parentCategory === '' ? null : parentCategory;
+        const categoryDoc = await Category.create({
+            name, 
+            parent: parent || undefined,
```

Comment:
Is this really a good idea to be able to insert into the database with undefined values??? Why not just reject it?